kong/globalpatches.lua

```diff
@@ -357,19 +357,16 @@ return function(options)
         local item = self.data[key]
         if item == nil then
           return nil, "not found"
-        else
-          local expire_at = item.info.expire_at
-          if expire_at == nil then
-            return 0
-          else
-            local remaining = expire_at - ngx.now()
-            if remaining < 0 then
-              return nil, "not found"
-            else
-              return remaining
-            end
-          end
         end
+        local expire_at = item.info.expire_at
+        if expire_at == nil then
+          return 0
+        end
+        local remaining = expire_at - ngx.now()
+        if remaining == 0 then
```

Comment:
<=0 ?

Reply:
THe ttl value could be negative, which means the expired time. I have tested this situation. 

But it seems that the[ lua-nginx-module doc](https://github.com/openresty/lua-nginx-module?tab=readme-ov-file#ngxshareddictttl) says that it only return nil if the item is expired. I think its a conflict between its doc and its real action.

Reply:
Interesting, TIL. In that case it might make sense to skip the `remaining == 0` check at all since it's not likely going to be hit.
It appears that the genuine shdict API only sweep the key if it's being `get` once.
```lua
local dict = ngx.shared.hello
dict:set("foo", "bar", 0.1)
ngx.sleep(0.2)
print(dict:ttl("foo"))
print(dict:get("foo"))
print(dict:ttl("foo"))
```
outputs:

-0.1

nilnot found

Reply:
Why i skip `remaining == 0` here is that `0` in[ lua-nginx-module doc ttl](https://github.com/openresty/lua-nginx-module?tab=readme-ov-file#ngxshareddictttl) means this key could exist forever, 

```
From https://github.com/openresty/lua-nginx-module?tab=readme-ov-file#ngxshareddictttl
A value of 0 means that the item will never expire.
```

So we must return `nil` instead of `0` to tell the caller that this key does not exist(expired).

And there is also a bug for `shdict:ttl()` in official lua-nginx-module, I have a reproduce to make this reproduce:

```
xc openresty $ cat ttl_0.sh
resty --shdict "hello 10m" -e '
local dict = ngx.shared.hello

dict:set("foo", "bar", 0.002)

for i = 1, 2 do
  ngx.sleep(0.001)
  local ttl = dict:ttl("foo")
  print(ttl)
  assert(ttl ~= 0)
end
'

 $ sh ttl_0.sh
0.001
0
ERROR: (command line -e):10: assertion failed!
stack traceback:
        (command line -e):10: in function 'inline_gen'
        init_worker_by_lua(nginx.conf:142):44: in function <init_worker_by_lua(nginx.conf:142):43>
        [C]: in function 'xpcall'
        init_worker_by_lua(nginx.conf:142):52: in function <init_worker_by_lua(nginx.conf:142):50>



```